mapboxcli/scripts/datasets.py

```diff
@@ -343,3 +345,146 @@ def create_tileset(ctx, dataset, tileset, name):
         click.echo(res.text)
     else:
         raise MapboxCLIException(res.text.strip())
+
+@datasets.command(name="put-features",
+    short_help="Add or update features in a dataset")
+@click.argument('dataset', required=True)
+@cligj.features_in_arg
+@click.pass_context
+def put_features(ctx, dataset, features):
+    """Insert or update features in a dataset.
+
+        $ mapbox dataset put-features features
+
+    FEATURES can be either the path to a file containing a FeatureCollection or
+    a line-delimited list of Feature objects. This function will make multiple
+    API requests in order to insert and/or update all the provided features.
+
+    All endpoints require authentication. An access token with
+    `datasets:write` scope is required, see `mapbox --help`.
+    """
+
+    service = ctx.obj.get('service')
+    to_put = []
+
+    def id():
```

Comment:
`id` is a python builtin - while python allows you to override builtins, we should probably just rename to `fid` to avoid conflicts.